regression/rt_gk_multib_sheath_1x2v_p1.c

```diff
@@ -0,0 +1,852 @@
+  }
+  else {
+    n = 0.5 * n_peak; // Ion total number density (right).
+  }
+
+  // Set ion total number density.
+  fout[0] = n;
+}
+
+void
+evalIonTempInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+
+  double Ti = app->Ti;
+
+  // Set ion isotropic temperature.
+  fout[0] = Ti;
+}
+
+void
+evalIonUparInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  // Set ion parallel velocity.
+  fout[0] = 0.0;
+}
+
+void
+evalIonSourceDensityInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+  double z = xn[0];
+
+  double n_src = app->n_src;
+  double Lz = app->Lz;
+
+  double n = 0.0;
+
+  if (fabs(z) < 0.25 * Lz) {
+    n = n_src; // Ion source total number density (left).
+  }
+  else {
+    n = 1.0e-40 * n_src; // Ion source total number density (right).
+  }
+
+  // Set ion source total number density.
+  fout[0] = n;
+}
+
+void
+evalIonSourceTempInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+
+  double T_src = app->T_src;
+
+  // Set ion source isotropic temperature.
+  fout[0] = T_src;
+}
+
+void
+evalIonSourceUparInit(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  // Set ion source parallel velocity.
+  fout[0] = 0.0;
+}
+
+void
+evalElcNu(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+
+  double nu_elc = app->nu_elc;
+
+  // Set electron collision frequency.
+  fout[0] = nu_elc;
+}
+
+void
+evalIonNu(double t, const double* GKYL_RESTRICT xn, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+
+  double nu_ion = app->nu_ion;
+
+  // Set ion collision frequency.
+  fout[0] = nu_ion;
+}
+
+static inline void
+mapc2p(double t, const double* GKYL_RESTRICT zc, double* GKYL_RESTRICT xp, void* ctx)
+{
+  // Set physical coordinates (X, Y, Z) from computational coordinates (x, y, z).
+  xp[0] = zc[0]; xp[1] = zc[1]; xp[2] = zc[2];
+}
+
+void
+bmag_func(double t, const double* GKYL_RESTRICT zc, double* GKYL_RESTRICT fout, void* ctx)
+{
+  struct sheath_ctx *app = ctx;
+
+  double B0 = app->B0;
+
+  // Set magnetic field strength.
+  fout[0] = B0;
+}
+
+struct gkyl_block_geom*
+create_block_geom(void *ctx)
+{
+
+  struct sheath_ctx *app = ctx;
+
+  struct gkyl_block_geom *bgeom = gkyl_block_geom_new(1, 3);
+
+  /* Block layout and coordinates
+
+   x  
+   ^  
+   |
+   1  +------------------+------------------+
+   |  |b0                |b1                |
+   |  |lower SOL         |upper             |
+   |  |                  |                  |
+   0  +------------------+------------------+
+      0 -----------------1------------------2----> z
+
```

Comment:
this diagram only has 2 blocks, but this input files has 3. Update the comment/diagram if that's correct.